### PR TITLE
Fixes #22015.  Integrate Derby 10.13.1.1 in Glassfish

### LINKED ISSUE #22015: Integrate Latest Derby in GlassFish
This task is for integrating Apache Derby 10.13.1.1 in Glassfish. The integration involves:

    1. Downloading the latest release version of Derby from Apache.
    2. Making changes in the relevant pom files to pick up the latest version.
    3. Update the app server/pom.xml and run the CI tests

### PR DESCRIPTION
**Pull request by pranjal10**
Fixes #22015 Integrate Derby 10.13.1.1 in Glassfish

### CODE REVIEW

**Comment by pranjal10** [comment]
@glassfishrobot Run CI tests please

**Comment by pranjal10** [comment]
@glassfishrobot Run CI tests please

File: appserver/packager/external/javadb/pom.xml [hunk 1]
```diff
@@ -51,31 +51,50 @@
...
     <name>JavaDB to be bundled into GlassFish</name>
+    
+    <properties>
+        <javadb.zip>${project.build.directory}/db-derby-${javadb.version}-bin.zip</javadb.zip>
+    </properties>
 
     <build>
         <plugins>
             <plugin>
+                <groupId>com.googlecode.maven-download-plugin</groupId>
```
Comment (romain-grecourt) [inline appserver/packager/external/javadb/pom.xml]: Using this plugin is a NO-NO.

**Comment by pranjal10** [comment]
@glassfishrobot Run CI tests please

**Comment by pranjal10** [comment]
@glassfishrobot Run CI tests please

**Comment by pranjal10** [comment]
@glassfishrobot Run CI tests please

**Comment by pranjal10** [comment]
@glassfishrobot Run CI tests please

**Comment by pranjal10** [comment]
@glassfishrobot Run CI tests please

**Comment by pranjal10** [comment]
@glassfishrobot Run CI tests please

**Comment by pranjal10** [comment]
@glassfishrobot Run CI tests please

**Review by yaminikb — APPROVED** [review]

**Pull request merged**